example-scripts/powershell/CherryPicking.ps1

```diff
@@ -0,0 +1,191 @@
+﻿#CherryPicking.ps1
+#Author: Daniel Siegl, LieberLieber Software GmbH
+#The purpose of this script is to start lemontree with a model from different branch in cherry pick mode!
+#Attention: the base and branchcopy of the file are collected from origin!
+#Supported:
+#.\example-scripts\powershell\CherryPicking.ps1 -Model DemoModel.eapx -Branch 70-test-for-discussion
+
+param
+(
+        [Parameter(Mandatory = $true)][string] $Model = "..\DemoModel.eapx",
+        [Parameter(Mandatory = $true)][string] $Branch = "70-test-for-discussion",
+        [Parameter(Mandatory = $false)][string] $ModelRootIds = $null#,
+        #for future use - filters are not supported by lemontree commandline - just by Session Files
+        #[Parameter(Mandatory = $false)][boolean] $conflicedFilter = 1 #if set to 1 it will add conflicted filters in the session
+)
+
+function Get-Fullname 
+{
+    param
+    (
+        [Parameter(Mandatory = $true)][string]$filename
+    )
+    process
+    {
+        if (Test-Path $filename) 
+        {
+            (Get-Item $fileName).FullName
+        } 
+    }
+}
+
+
+#Check if the script is called from inside a git repo return -1 one on failure
+function isGit
+{
+    process
+    {
+        $isGit = git rev-parse --is-inside-work-tree
+        if (!$isGit)
+        {
+            echo "This script needs to be called from inside a git repo directory."
+            exit -1
+        }
+    }
+}
+
+#convert the "windows path to a git path - works with realtive and absolute path. return -1 one on failure
+function Get-GitMappedFilePaths  
+{
+    param
+    (
+        [Parameter(Mandatory = $true)][string] $filename
+    )
+    process
+    {
+        $gitFilename = git ls-files --full-name $filename
+        if ([string]::IsNullOrEmpty($gitFilename))
+        {
+            echo "This filename parameter doesn't point to an existing file inside The repo directory."
+            exit -1
+        }
+        return @($gitFilename);
+    }
+}
+
+#Get the files from the remotes by commitid via http and curl - no LFS or not LFS handling needed
+function Get-DownloadedFilename  
+{
+    param
+    (
+        [Parameter(Mandatory = $true)][string] $gitFilename,
+        [Parameter(Mandatory = $true)][string] $commitID
+    )
+    process
+    {
+        $gitUri =  git config --get remote.origin.url
+        $gitUri = $gitUri.Replace(".git","");
+        $stripPathFilename = $commitID + "_"+[System.IO.Path]::GetFileName($gitFilename)
+        $tempFilename = Join-Path -Path "$env:TEMP" -ChildPath "$stripPathFilename"
+        $url = "$gitUri/raw/$commitID/$gitFilename"
```

Comment:
This is only working on github but not on other git servers with other LFS integrations
Invalid request from host x.x.x.x: Note that Bitbucket does not support Git's legacy HTTP transport protocol.